### DISCUSSION TITLE
Default value for dynamic <option> Elements ignored

### DISCUSSION THREAD

**Discussion by mo86** [alpinejs/alpine#3405] [7. Bugs (archived)]
Hi,

I got an array with dynamic options and I would like to set the default selected value/option but it doesnt work. In the following example I would assume that Option Two is preselected but it isn't.

```
<div x-data="{selectedValue:2, options:[{label:'Option One',value:1},{label:'Option Two',value:2}]}">
    <select name="xxx" x-model="selectedValue">
        <template x-for="(option, index) in options">
            <option :value="option.value" x-text="option.label"></option>
        </template>
    </select>
</div>
```
Thanks for your help :)

**Comment by SimoTod** [comment]
The x-model tag gets processed before the options exist so it will never update the select on load. You need to add `:selected="option.value === selectedValur"`

**Reply by mo86** [reply]
Thank you @SimoTod

**Reply by mo86** [reply]
I understand but I am wondering why it also doesn't work when I change "selectedValue" afterwards to a different value. For example when I add a button which changes the value on click. So it seems like there is no connection between the property and the the template options.

**Reply by SimoTod** [reply]
It works for me: https://codepen.io/SimoTod/pen/yLqrVMO

**Comment by mo86** [comment]
Oh yes you are right. I will use your solution which feels a little bit like a hack. It would be nice to have some "lifecycle" events like "mounted" to change the value when all alpine components are ready :)

**Reply by SimoTod** [reply]
X-init + $nextTick does that

**Reply by SimoTod** [reply]
`x-init="$nextTick(() => selectedValue = 1)"` but I find it less immediate than declaring an option as swlected

**Reply by ekwoka** [reply]
@SimoTod What do you think about having Alpine recognize when it's binding on a select and next ticking itself?

**Reply by SimoTod** [reply]
Probably fine, it could even always use nextTick and nobody would notice the difference, I suspect

**Reply by mo86** [reply]
Guys thank you very much for your answers. Helped me a lot. I wanted to make sure I don't have a bug in the code and there is no out of the box solution for this :)
